### DISCUSSION TITLE
No Audio from HDMI on C64

### DISCUSSION THREAD

**Discussion by michaelniotis** [frntc/Sidekick64#52] [Q&A]
Trying to figure out why there is no Audio output from the HDMI port of the RPi. The only thing I am getting is the SideKick splash screen. The Settings is also set for HDMI audio out. I have Trying multiple sid options.

Is there any way to fix this?

**Comment by frntc** [comment]
just to make sure: you've enabled the SID/FM/MIDI emulation prior to launching a program (F3 in main menu)? What HDMI device did you connect?

**Reply by michaelniotis** [reply]
Correct. Also the Jumper cable is installed on the SID Chip. Otherwise, SK64 Complains that there is no SID CS Signal. The Device is a Portable monitor with integrated speakers. I also added "hdmi_drive=2" in the Config file to see if it makes a difference. BTW This was on a RPI 3B. Otherwise everything else was working fine.

I tried this on 2 Devices with the same result (No HDMI Audio)

**Comment by frntc** [comment]
Do you have the SID-emulation-animation on your TFT like in this video https://www.youtube.com/watch?v=0uilbNnjCPo ? (you said splash screen, so I'm not sure what you meant)

**Reply by michaelniotis** [reply]
Yes. There is waveform animation on the TFT screen. The HDMI output is only showing the SideKick64 Logo and nothing else. Is it supposed to show only that on the C64/C128 since there is no VIC emulation?

Also when playing the Tuneful eight for example I only hear some garbled sounds from the SID but nothing from the HDMI. Also modplay plays the Audio from the SID chip.

**Comment by frntc** [comment]
it's normal that you hear garbled sounds from the internal SID when running Tuneful 8, that's not a SK64-issue.

IIRC you have to press SHIFT+RETURN to play a MOD-file via HDMI. 

But I suspect that the problem is rather somewhere else. If you chose HDMI out for audio (it's correct that the HDMI screen only shows the logo) there should be sound. Have you tried if there is audio if you boot Linux on your RPi?

**Reply by michaelniotis** [reply]
OK. I just got the time to test. So MOD play over HDMI seems to be working fine with the SHIFT+Return Key. I also tried a retropie SD Card and audio over HDMI also works on the same device with the sidekick hat also attached.

It seems it is only the SID/FM emulation that doesn't seem to output any Audio at all over HDMI.

**Reply by frntc** [reply]
The MOD-player and SID-emulation both use the same HDMI-device/code, so if one works, the other should do as well (are you really sure you chose HDMI out in the F5-config screen? maybe you forget to save your settings?). Are you using a RPi 3A+/B+? Then you could check if the sound is on the jack plug.

**Comment by blindguy** [comment]
Having the same kinda issue.  When I play mods via HDMI it is fine but for SID I have to use a external speaker connected to the 3.5 mm jack and the audio is very low on that.  I"m going to try and turn the volump ALL the way up on the HDMI to see if the problem is just low volume on the HDMI via SID.  
I"m using a 3B+ everything else seems to be working fine.
Thanks!

**Reply by frntc** [reply]
The audio jack PWM is known to have low volume. For HDMI (assuming that the SID-emulation shows its visualization on the TFT): can you try another HDMI device?

**Reply by blindguy** [reply]
I tried a few other HDMI devices and still no audio.  Again, the HDMI audio does work fine when I use the shift+enter in the music section for Mods and such, just not when I attempt to use the 8 SIDS.  Any other suggestions would be appreciated as I love the cart!

**Reply by frntc** [reply]
I found the HDMI-bug, fix will be included in the next release! (Accidentally muted output, but still not every HDMI-device works equally well and that's nothing I can change)

**Reply by GameTimeWarp** [reply]
Excellent news! In the meantime I have been using v0.5 and HDMI output is beautiful :)

**Reply by blindguy** [reply]
> I found the HDMI-bug, fix will be included in the next release! (Accidentally muted output, but still not every HDMI-device works equally well and that's nothing I can change)

Used the new v.5 version and that fixed the issues for me!!! Sounds fantastic!!!
